### PR TITLE
Stricter TS config; fix implicit any

### PR DESCRIPTION
**Pull request by freebroccolo**
Just for reference, details about the specific TS config options can be found [here](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

### CODE REVIEW

**Comment by freebroccolo** [comment]
Let me know if you have any questions about these changes. The interface types I added were synthesized from type checker then cleaned up.

The changes to the typescript config are the same as for the recent PR I made to the `core` repo.

Mostly it sorts the keys, changes the target to `es2018`, and makes the defaults stricter so that fewer `any` types are allowed.

File: src/check.ts [hunk 1]
```diff
@@ -319,14 +339,14 @@ See https://github.com/actions-rs/clippy-check/issues/2 for details.`);
     /// Convert parsed JSON line into the GH annotation object
     ///
     /// https://developer.github.com/v3/checks/runs/#annotations-object
-    static makeAnnotation(contents): octokit.ChecksCreateParamsOutputAnnotations {
-        const primarySpan = contents.message.spans.find((span) => span.is_primary == true);
+    static makeAnnotation(contents: AnnotationContents): octokit.ChecksCreateParamsOutputAnnotations {
+        const primarySpan: undefined | MessageSpan = contents.message.spans.find((span) => span.is_primary == true);
```
Comment (svartalf) [inline src/check.ts]: Should not the type here be inferred automatically?

File: src/check.ts [hunk 2]
```diff
@@ -7,6 +7,26 @@ import {plural} from './render';
 
 const USER_AGENT = `${pkg.name}/${pkg.version} (${pkg.bugs.url})`;
 
+interface AnnotationContents {
+    reason?: string,
```
Comment (svartalf) [inline src/check.ts]: I think that this should be a required field

File: src/check.ts [hunk 3]
```diff
@@ -7,6 +7,26 @@ import {plural} from './render';
...
+    message: {
+        code: string,
+        level: string,
+        message: string,
+        rendered: string,
+        spans: MessageSpan[],
+    },
+}
+
+interface MessageSpan {
```
Comment (svartalf) [inline src/check.ts]: Do you mind to rename it into the `DiagnosticSpan` to match the rust version: https://github.com/rust-lang/rust/blob/b7820b28937ba875521a2525cf75223c80faf9f2/src/libsyntax/json.rs#L141 ?

File: src/check.ts [hunk 4]
```diff
@@ -7,6 +7,26 @@ import {plural} from './render';
 
 const USER_AGENT = `${pkg.name}/${pkg.version} (${pkg.bugs.url})`;
 
+interface AnnotationContents {
```
Comment (svartalf) [inline src/check.ts]: This one is actually called `Message` at [cargo sources](https://github.com/rust-lang/cargo/blob/d764fff2da618b24fbf506229509e3d183050c2e/src/cargo/util/machine_message.rs#L9), maybe it is worth to rename it into smth like `CargoMessage`, as it is not an annotation yet?

File: src/check.ts (hunk 1 above)
Comment (freebroccolo) [reply src/check.ts]: Yeah, I'm not sure why in this case but in general typescript has trouble inferring the types of arguments I've found. You can check in the vscode editor mode by hovering over the argument. In this case it should show `contents: any` I think.

**Comment by freebroccolo** [comment]
Thanks for the feedback!

**Comment by svartalf** [comment]
No, thank you for making it better :)

**Pull request merged**